### PR TITLE
Feature/separate interfaces

### PR DESCRIPTION
**Pull request by hroest**
# Description

This is a subset of the changes from #5613 that I made when trying to separate OpenMS into different libraries. The main thing this PR does is to remove the dependency of various parts of OpenMS on xercesc which gets pulled in with `XMLHandler`. Unfortunately, including `XMLHandler.h` is quite common in OpenMS which leads to a large number of cross-dependencies and prevents us from using clean interfaces to interact with files. 

One part of this PR is to introduce a new interface for FeatureXML and ConsensusXML and move the implementation into the `OpenMS::Internal` namespace

Secondly, it uses the `FileHandler` as a clean interface to load and write files

# Checklist:
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

### CODE REVIEW

**Comment by jpfeuffer** [comment]
Very nice. Is this enough to completely hide Xerces from the interface or is there more to be done?

The change in loading might break a lot of pyopenms workflows but in my opinion it is worth it. We probably should do an increase in the major version then.

**Comment by hroest** [comment]
> Very nice. Is this enough to completely hide Xerces from the interface or is there more to be done?

there is more to be done, this is just a first step for the worst offenders:

```
$ grep XMLHandler.h OpenMS/FORMAT/*.h
OpenMS/FORMAT/CVMappingFile.h:#include <OpenMS/FORMAT/HANDLERS/XMLHandler.h>
OpenMS/FORMAT/IdXMLFile.h:#include <OpenMS/FORMAT/HANDLERS/XMLHandler.h>
OpenMS/FORMAT/MascotXMLFile.h:#include <OpenMS/FORMAT/HANDLERS/MascotXMLHandler.h>
OpenMS/FORMAT/OMSSAXMLFile.h:#include <OpenMS/FORMAT/HANDLERS/XMLHandler.h>
OpenMS/FORMAT/PepXMLFile.h:#include <OpenMS/FORMAT/HANDLERS/XMLHandler.h>
OpenMS/FORMAT/PepXMLFileMascot.h:#include <OpenMS/FORMAT/HANDLERS/XMLHandler.h>
OpenMS/FORMAT/ProtXMLFile.h:#include <OpenMS/FORMAT/HANDLERS/XMLHandler.h>
OpenMS/FORMAT/QcMLFile.h:#include <OpenMS/FORMAT/HANDLERS/XMLHandler.h>
OpenMS/FORMAT/TransformationXMLFile.h:#include <OpenMS/FORMAT/HANDLERS/XMLHandler.h>
OpenMS/FORMAT/XTandemXMLFile.h:#include <OpenMS/FORMAT/HANDLERS/XMLHandler.h>
```

basically we would have to do one of two things for all of these: write wrapper functions in FileHandler or separate implementation (with XMLHandler) from interface. Quite a bit of work but it should also help with #5619 and make sure that downstream consumers dont pull in stuff they dont need/want. 

> The change in loading might break a lot of pyopenms 

as long as we keep the old interfaces around, this should not actually be a problem

File: src/openms/source/ANALYSIS/DECHARGING/FeatureDeconvolution.cpp [hunk 1]
```diff
@@ -1039,8 +1039,7 @@ namespace OpenMS
     FeatureMapType fm_missing;
     cl.suggestMissingFeatures(fm_out, cons_map, fm_missing);
 
-    FeatureXMLFile fmf;
-    fmf.store("fm_missing.featureXML", fm_missing);
+    FileHandler.storeFeatures("fm_missing.featureXML", fm_missing);
```
Comment (timosachsenberg) [inline src/openms/source/ANALYSIS/DECHARGING/FeatureDeconvolution.cpp:1042]: maybe worth to dicuss: is it a better API for a user to have this moved to a FileHandler class? Wouldn't it be more natural to have such a method in FeatureMap? Separation is still be possible if I am not mistaken.

Comment (timosachsenberg) [reply src/openms/source/ANALYSIS/DECHARGING/FeatureDeconvolution.cpp:1042]: or e.g., keeping the old interface with FeatureXMLFile etc. that uses some internal FeatureXMLFileHandler

Comment (hroest) [reply src/openms/source/ANALYSIS/DECHARGING/FeatureDeconvolution.cpp:1042]: > e.g., keeping the old interface with FeatureXMLFile, so code doesn't break and call some internal FeatureXMLFileHandler

this is what we do for `MzMLFile` and `MzMLFileHandler` and what I implemented here in this PR as well, separating the interface `FeatureXMLFile` and `FeatureXMLFileHandler`. But the question that makes sense is whether we need 2 interfaces, `FileHandler` and `MzMLFile`

Comment (timosachsenberg) [reply src/openms/source/ANALYSIS/DECHARGING/FeatureDeconvolution.cpp:1042]: Ok I see. I agree - having a FeatureXMLFile interface looks a bit cleaner designwise compared to having only a "FileHandler" class. But no strong opinion here.

Comment (timosachsenberg) [reply src/openms/source/ANALYSIS/DECHARGING/FeatureDeconvolution.cpp:1042]: But let's keep the two interfaces for now

**Comment by hroest** [comment]
@timosachsenberg per cdash all tests pass and builds work: http://cdash.openms.de/index.php?project=OpenMS&date=2021-10-29 should we merge?

**Comment by timosachsenberg** [comment]
rebuild jenkins

**Comment by timosachsenberg** [comment]
Thanks!

**Pull request merged**